remoteworkitem/flatten.go

```diff
@@ -0,0 +1,56 @@
+package remoteworkitem
+
+import (
+	"reflect"
+	"strings"
+)
+
+type stack []string
+
+func (s *stack) push(v string) {
+	*s = append(*s, v)
+}
+
+func (s *stack) pop() string {
+	l := len(*s)
+	if l == 0 {
+		return ""
+	}
+	stackReference := *s // because *s[:l-1] did not work.
+	poppedData := stackReference[l-1]
+	*s = stackReference[:l-1]
+	return poppedData
+}
+
+func (s stack) String() string {
+	return strings.Join(s, ".")
+}
+
+// MapFlattener defines the data structures and the implemetation of how to flatten a nested map.
+type MapFlattener struct {
+	nestedKeyStack stack
+	flattendMap    map[string]interface{}
+}
+
+// traverseMapAsDFS does a depth-first traversal
+func (fm MapFlattener) traverseMapAsDFS(value map[string]interface{}) {
```

Comment:
I've seen it being done in a few places but I'm not sure why we need to use something like a 'call-by-reference' if the internal state change is working. Would be an interesting conversation to me :)